monai/apps/nnunet/nnunetv2_runner.py

```
@@ -525,35 +525,70 @@ def train_single_model(self, config: Any, fold: int, gpu_id: tuple | list | int
+        Raises:
+            ValueError: If gpu_id is an empty tuple or list.
+        """
+        env = os.environ.copy()
+        device_setting: str | None = None
+        num_gpus = 1
+        if isinstance(gpu_id, str):
+            device_setting = gpu_id
+            num_gpus = 1
+        elif isinstance(gpu_id, (tuple, list)):
+            if len(gpu_id) == 0:
+                raise ValueError("gpu_id tuple/list cannot be empty")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)